src/components/main-view/MainView.jsx

```diff
@@ -20,44 +20,32 @@ export const MainView = () => {
     fetch("https://strobeapp-583fefccfb94.herokuapp.com/movies", {
       headers: { Authorization: `Bearer ${token}` }
     })
-      .then((response) => response.json())
+      .then((response) => {
+        if (!response.ok) throw new Error("Failed to fetch movies");
+        return response.json();
+      })
       .then((data) => {
-        const moviesFromApi = data.map((movie) => ({
-          _id: movie._id,
-          Title: movie.Title,
-          Description: movie.Description,
-          Genre: movie.Genre,
-          Director: movie.Director,
-          ImagePath: movie.ImagePath,
-          Featured: movie.Featured,
-        }));
-        setMovies(moviesFromApi);
+        setMovies(data.map(({ _id, Title, Description, Genre, Director, ImagePath, Featured }) => ({
+          _id, Title, Description, Genre, Director, ImagePath, Featured
+        })));
```

Comment:
Hey @Dbreazi ,

It’s not mandatory to use the `.map` function, and in this particular case, it’s redundant since you’re not modifying the keys.

```suggestion
        setMovies(data);
```